Duplicati/CommandLine/Commands.cs

```diff
@@ -234,14 +235,86 @@ public static int Affected(TextWriter outwriter, Action<Duplicati.Library.Main.C
-            return argList.Select(arg => arg.IndexOfAny(specialArgChars) < 0
-                 && !arg.StartsWith("[", StringComparison.Ordinal) ? "*" + Util.DirectorySeparatorString + arg : arg).ToList();
+            var containsSeparators = ContainsDirectorySeparators(arg);
+            var containsWildcards = ContainsWildcards(arg);
+
+            if (!containsSeparators && arg.StartsWith("@", StringComparison.Ordinal))
+            {
+                // Convert to Regexp filter and prefix with ".*/"
+                return $"[.*{Regex.Escape(Util.DirectorySeparatorString + arg.Substring(1))}]";
+            }
+            else if (!containsSeparators && !containsWildcards && !arg.StartsWith("[", StringComparison.Ordinal))
+            {
+                // Prefix files with "*/"
+                return "*" + Util.DirectorySeparatorString + arg;
+            }
+            else
+            {
+                return arg;
+            }
+        }
+
+        /// <summary>
+        /// For folders, replace argument with the equivalent of suffixing
+        /// with "*" so we restore contents in the folder.
+        /// </summary>
+        public static IEnumerable<string> SuffixArgsWithAsterisk(IEnumerable<string> argList) =>
+            argList.Select(SuffixArgWithAsterisk);
+
+        /// <summary>
+        /// For folders, return argument with the equivalent of suffixing
+        /// with "*" so we restore contents in the folder.
+        /// </summary>
+        private static string SuffixArgWithAsterisk(string arg)
+        {
+            var containsWildcards = ContainsWildcards(arg);
+            var endsWithSeparator = arg.EndsWith(Util.DirectorySeparatorString);
+
+            if (endsWithSeparator && arg.StartsWith("@", StringComparison.Ordinal))
+            {
+                // Convert to Regexp filter and suffix with ".*"
+                return $"[{Regex.Escape(arg.Substring(1))}.*]";
+            }
+            else if (endsWithSeparator && !containsWildcards && !arg.StartsWith("[", StringComparison.Ordinal))
+            {
+                // Suffix with "*"
+                return arg + "*";
+            }
+            else
+            {
+                return arg;
+            }
         }
 
+        private static readonly char[] pathSeparatorsCharacters = new[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar };
+
+        /// <summary>
+        /// Returns true if <paramref name="s"/> contains directory separator characters.
+        /// </summary>
+        public static bool ContainsDirectorySeparators(string s) =>
+            s.IndexOfAny(pathSeparatorsCharacters) >= 0;
+
+        private static readonly char[] wildcardCharacters = new[] { '*', '?' };
+
+        /// <summary>
+        /// Returns true if <paramref name="s"/> contains wildcard characters.
+        /// </summary>
+        public static bool ContainsWildcards(string s) =>
+            s.IndexOfAny(wildcardCharacters) >= 0;

```

Comment:
I think this is short enough where we don't need the line break.

Reply:
Fixed in latest commit.